File: webapp/src/app/dto/accessibility.dto.ts
import { Accessibility } from '../interface/accessibility.interface';

export class AccessibilityDTO {
  constructor(
    private _id: string,
    private _toiletGrabBarAvailable: boolean,
    private _showerGrabBarAvailable: boolean,
    private _stepFreeShowerAvailable: boolean,
    private _showerBathChairAvailable: boolean,
    private _stepFreeBedroomAccessAvailable: boolean,
    private _wideBedroomEntranceAvailable: boolean,
    private _stepFreeAccessAvailable: boolean
  ) {}

  public get id(): string {
    return this._id;
  }

  public set id(id: string) {
    this._id = id;
  }

  public get toiletGrabBarAvailable(): boolean {
    return this._toiletGrabBarAvailable;
  }

  public set toiletGrabBarAvailable(toiletGrabBarAvailable: boolean) {
    this._toiletGrabBarAvailable = toiletGrabBarAvailable;
  }

  public get showerGrabBarAvailable(): boolean {
    return this._showerGrabBarAvailable;
  }

  public set showerGrabBarAvailable(showerGrabBarAvailable: boolean) {
    this._showerGrabBarAvailable = showerGrabBarAvailable;
  }

  public get stepFreeShowerAvailable(): boolean {
    return this._stepFreeShowerAvailable;
  }

  public set stepFreeShowerAvailable(stepFreeShowerAvailable: boolean) {
    this._stepFreeShowerAvailable = stepFreeShowerAvailable;
  }

  public get showerBathChairAvailable(): boolean {
    return this._showerBathChairAvailable;
  }

  public set showerBathChairAvailable(showerBathChairAvailable: boolean) {
    this._showerBathChairAvailable = showerBathChairAvailable;
  }

  public get stepFreeBedroomAccessAvailable(): boolean {
    return this._stepFreeBedroomAccessAvailable;
  }

  public set stepFreeBedroomAccessAvailable(
    stepFreeBedroomAccessAvailable: boolean
  ) {
    this._stepFreeBedroomAccessAvailable = stepFreeBedroomAccessAvailable;
  }

  public get wideBedroomEntranceAvailable(): boolean {
    return this._wideBedroomEntranceAvailable;
  }

  public set wideBedroomEntranceAvailable(
    wideBedroomEntranceAvailable: boolean
  ) {
    this._wideBedroomEntranceAvailable = wideBedroomEntranceAvailable;
  }

  public get stepFreeAccessAvailable(): boolean {
    return this._stepFreeAccessAvailable;
  }

  public set stepFreeAccessAvailable(stepFreeAccessAvailable: boolean) {
    this._stepFreeAccessAvailable = stepFreeAccessAvailable;
  }

  public static fromInterface(accessibility: Accessibility): AccessibilityDTO {
    return new AccessibilityDTO(
      accessibility.id,
      accessibility.toiletGrabBarAvailable,
      accessibility.showerGrabBarAvailable,
      accessibility.stepFreeShowerAvailable,
      accessibility.showerBathChairAvailable,
      accessibility.stepFreeBedroomAccessAvailable,
      accessibility.wideBedroomEntranceAvailable,
      accessibility.stepFreeAccessAvailable
    );
  }
}

File: webapi/src/main/java/dev/hekmyr/olids/api/dto/AccessibilityDTO.java
package dev.hekmyr.olids.api.dto;

import dev.hekmyr.olids.api.entity.Accessibility;
import java.util.UUID;

public class AccessibilityDTO {

  private UUID id;
  private boolean toiletGrabBarAvailable;
  private boolean showerGrabBarAvailable;
  private boolean stepFreeShowerAvailable;
  private boolean showerBathChairAvailable;
  private boolean stepFreeBedroomAccessAvailable;
  private boolean wideBedroomEntranceAvailable;
  private boolean stepFreeAccessAvailable;

  public AccessibilityDTO() {}

  public AccessibilityDTO(Accessibility model) {
    this.id = model.getId();
    this.toiletGrabBarAvailable = model.isToiletGrabBarAvailable();
    this.showerGrabBarAvailable = model.isShowerGrabBarAvailable();
    this.stepFreeShowerAvailable = model.isStepFreeShowerAvailable();
    this.showerBathChairAvailable = model.isShowerBathChairAvailable();
    this.stepFreeBedroomAccessAvailable =
      model.isStepFreeBedroomAccessAvailable();
    this.wideBedroomEntranceAvailable =
      model.isWideBedroomEntranceAvailable();
    this.stepFreeAccessAvailable = model.isStepFreeAccessAvailable();
  }

  public UUID getId() {
    return id;
  }

  public void setId(UUID id) {
    this.id = id;
  }

  public boolean isToiletGrabBarAvailable() {
    return toiletGrabBarAvailable;
  }

  public void setToiletGrabBarAvailable(boolean toiletGrabBarAvailable) {
    this.toiletGrabBarAvailable = toiletGrabBarAvailable;
  }

  public boolean isShowerGrabBarAvailable() {
    return showerGrabBarAvailable;
  }

  public void setShowerGrabBarAvailable(boolean showerGrabBarAvailable) {
    this.showerGrabBarAvailable = showerGrabBarAvailable;
  }

  public boolean isStepFreeShowerAvailable() {
    return stepFreeShowerAvailable;
  }

  public void setStepFreeShowerAvailable(boolean stepFreeShowerAvailable) {
    this.stepFreeShowerAvailable = stepFreeShowerAvailable;
  }

  public boolean isShowerBathChairAvailable() {
    return showerBathChairAvailable;
  }

  public void setShowerBathChairAvailable(boolean showerBathChairAvailable) {
    this.showerBathChairAvailable = showerBathChairAvailable;
  }

  public boolean isStepFreeBedroomAccessAvailable() {
    return stepFreeBedroomAccessAvailable;
  }

  public void setStepFreeBedroomAccessAvailable(
    boolean stepFreeBedroomAccessAvailable
  ) {
    this.stepFreeBedroomAccessAvailable = stepFreeBedroomAccessAvailable;
  }

  public boolean isWideBedroomEntranceAvailable() {
    return wideBedroomEntranceAvailable;
  }

  public void setWideBedroomEntranceAvailable(
    boolean wideBedroomEntranceAvailable
  ) {
    this.wideBedroomEntranceAvailable = wideBedroomEntranceAvailable;
  }

  public boolean isStepFreeAccessAvailable() {
    return stepFreeAccessAvailable;
  }

  public void setStepFreeAccessAvailable(boolean stepFreeAccessAvailable) {
    this.stepFreeAccessAvailable = stepFreeAccessAvailable;
  }
}
